POSIX/posix-cflags.sh

```diff
@@ -21,6 +21,24 @@ if ! ${CC} -D_POSIX_C_SOURCE=200809L posix-clock_realtime.c 2>/dev/null; then
 	printf %s "-DPOSIXFAIL_CLOCK_REALTIME"
 	echo "WARNING: POSIX violation: <time.h> not defining CLOCK_REALTIME" 1>&2
 fi
+rm -f posix-clock_gettime
```

Comment:
Why deleting the file here when we're about to write over it anyway?

Reply:
It seemed more reliable that way.  What guarantees do we have about `c99` compilers faced with an error?

- if `gcc` (or `clang`) attempts to compile `/dev/null`, they delete `a.out`.
- if they fail because of a wrong header (or some random junk in the `.c` file), they don't touch `a.out`.

Granted, I suppose that we only run the binary if the compilation succeeds, and in that case, we're guaranteed that the output is fresh.  Hmm, ok.